--- gdk-pixbuf-2.26.1/gdk-pixbuf/Makefile.am.orig	2012-04-14 20:11:34.000000000 +0200
+++ gdk-pixbuf-2.26.1/gdk-pixbuf/Makefile.am	2012-04-15 18:03:52.776039347 +0200
@@ -208,7 +208,7 @@
 # create a partial one that's enough for our use.
 
 libgdiplus = libgdiplus.dll.a
-gdiplus_ldflag = -Wl,$(libgdiplus)
+gdiplus_ldflag = -L. -lgdiplus # -Wl,$(libgdiplus)
 
 libgdiplus.dll.a: gdiplus.def
 	$(DLLTOOL) --kill-at --dllname gdiplus.dll --input-def $(srcdir)/gdiplus.def --output-lib $@
@@ -265,7 +265,7 @@
 	libpixbufloader-gdip-jpeg.la \
 	libpixbufloader-gdip-tiff.la
 
-libpixbufloader_gdip_ico_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_ico_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_ico_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -274,9 +274,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-ico.c
-libpixbufloader_gdip_ico_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_ico_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_wmf_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_wmf_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_wmf_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -285,9 +285,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-wmf.c
-libpixbufloader_gdip_wmf_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_wmf_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_emf_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_emf_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_emf_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -296,9 +296,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-emf.c
-libpixbufloader_gdip_emf_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_emf_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_bmp_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_bmp_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_bmp_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -307,9 +307,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-bmp.c
-libpixbufloader_gdip_bmp_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_bmp_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_gif_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_gif_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_gif_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -318,9 +318,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-gif.c
-libpixbufloader_gdip_gif_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_gif_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_jpeg_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_jpeg_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_jpeg_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -329,9 +329,9 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-jpeg.c
-libpixbufloader_gdip_jpeg_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_jpeg_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
-libpixbufloader_gdip_tiff_la_LDFLAGS = -avoid-version -module -no-undefined $(gdiplus_ldflag)
+libpixbufloader_gdip_tiff_la_LDFLAGS = -avoid-version -module -no-undefined
 libpixbufloader_gdip_tiff_la_SOURCES = 	\
 	io-gdip-native.h		\
 	io-gdip-propertytags.h		\
@@ -340,7 +340,7 @@
 	io-gdip-animation.c		\
 	io-gdip-animation.h		\
 	io-gdip-tiff.c
-libpixbufloader_gdip_tiff_la_LIBADD = $(module_libs) $(libole32)
+libpixbufloader_gdip_tiff_la_LIBADD = $(module_libs) $(libole32) $(gdiplus_ldflag)
 
 endif
 
