--- gdk-pixbuf-2.30.0/tests/pixbuf-randomly-modified.c.orig	2013-09-23 15:06:17.000000000 +0200
+++ gdk-pixbuf-2.30.0/tests/pixbuf-randomly-modified.c	2013-09-24 20:19:45.937337489 +0200
@@ -26,7 +26,9 @@
 #include <time.h>
 #include <string.h>
 #include <sys/time.h>
+#ifndef __WIN32__
 #include <sys/resource.h>
+#endif
 
 static void
 disaster (const char *what)
@@ -104,12 +106,14 @@
   gboolean got_seed = FALSE;
   GPtrArray *files = g_ptr_array_new ();
   int l, iterations;
+#ifndef __WIN32__
   struct rlimit max_mem_size;
 
   max_mem_size.rlim_cur = 100 * 1024 * 1024; /* 100M */
   max_mem_size.rlim_max = max_mem_size.rlim_cur;
   setrlimit (RLIMIT_DATA, &max_mem_size);
   setrlimit (RLIMIT_AS, &max_mem_size);
+#endif
 
   g_test_init (&argc, &argv, NULL);
 
